File: wlr/OGUR/OGUR/Entities/Upstairs.cs
using OGUR.Creatures;
using OGUR.Dungeons;
using SPX.Entities;
using OGUR.Management;
using SPX.Core;
using System;
using System.Runtime.Serialization;

namespace OGUR.Entities
{
    [Serializable()]
    public class Upstairs : Entity
    {
        public Upstairs(SerializationInfo info, StreamingContext context):base(info, context){}
        public Upstairs(Point2 location)
        {
            Initialize(location, SpriteType.UPSTAIRS, OGUR.EntityType.UPSTAIRS,ZDepth.Stairs);
        }
        public int GetTargetLocation()
        {
            return Location.Depths;
        }

        ICreature player;
        public override void  Update()
        {
            player = EntityManager.GetTouchingCreature(this) as Player;
            if(player!=null)
            {
                if (player.IsInteracting())
                {
                    player.PerformInteraction();
                    DungeonFactory.GetPreviousFloor(GetTargetLocation());
                }    
            }
        }
    }
}

File: proj/src/aigilas/entities/Upstairs.java
package aigilas.entities;

import aigilas.creatures.BaseCreature;
import aigilas.dungeons.DungeonFactory;
import aigilas.management.SpriteType;
import spx.bridge.DrawDepth;
import spx.bridge.EntityType;
import spx.core.Point2;
import spx.entities.Entity;
import spx.entities.EntityManager;

public class Upstairs extends Entity {
    public Upstairs(Point2 location) {
        initialize(location, SpriteType.UPSTAIRS, EntityType.UPSTAIRS, DrawDepth.Stairs);
    }

    BaseCreature player;

    @Override
    public void update() {
        player = (BaseCreature) EntityManager.getTouchingCreature(this);
        if (player != null) {
            if (player.isInteracting()) {
                player.performInteraction();
                try {
                    DungeonFactory.getPreviousFloor();
                } catch (Exception e) {
                    e.printStackTrace();
                }
            }
        }
    }
}
